File: Core/src/main/java/com/hqq/core/utils/GsonUtil.java
package com.hqq.core.utils;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.JsonElement;
import com.google.gson.JsonParser;
import com.google.gson.JsonSyntaxException;

import java.lang.reflect.Type;

/**
 * @Author : huangqiqiang
 * @Package : com.qi.core.utils
 * @FileName :   GsonUtil
 * @Date : 2019/1/3 0003  上午 10:30
 * @Descrive :
 * @Email :  qiqiang213@gmail.com
 */
public class GsonUtil {
    private static final Gson GSON = new GsonBuilder()
            .serializeNulls()//排除transient标记的字段
            .create();

    public static String toJsonString(Object o) {
        return GSON.toJson(o);
    }

    public static String toJsonString(Object o, Type type) {
        return GSON.toJson(o, type);
    }

    public static <T> T fromJson(String json, Class<T> clazz) {
        try {

            return GSON.fromJson(json, clazz);
        } catch (JsonSyntaxException e) {
            return null;
        }
    }

    public static <T> T fromJson(String json, Type type) {
        try {
            return GSON.fromJson(json, type);
        } catch (JsonSyntaxException e) {
            return null;
        }
    }

    public static <T> T fromJson(JsonElement json, Class<T> clazz) {
        try {
            return GSON.fromJson(json, clazz);
        } catch (JsonSyntaxException e) {
            return null;
        }
    }

    public static <T> T fromJson(JsonElement json, Type type) {
        try {
            return GSON.fromJson(json, type);
        } catch (JsonSyntaxException e) {
            return null;
        }
    }

    public static JsonElement toJsonTree(Object o) {
        return GSON.toJsonTree(o);
    }

    public static JsonElement toJsonTree(Object o, Type type) {
        return GSON.toJsonTree(o, type);
    }

    public static JsonElement getValue(String json, String key) {
        return new JsonParser().parse(json).getAsJsonObject().get(key);
    }

}

File: Core/src/main/java/com/hqq/core/utils/GsonUtil.kt
package com.hqq.core.utils

import com.google.gson.GsonBuilder
import com.google.gson.JsonElement
import com.google.gson.JsonParser
import com.google.gson.JsonSyntaxException
import java.lang.reflect.Type

/**
 * @Author : huangqiqiang
 * @Package : com.qi.core.utils
 * @FileName :   GsonUtil
 * @Date : 2019/1/3 0003  上午 10:30
 * @Email :  qiqiang213@gmail.com
 * @Descrive :
 */
object GsonUtil {
    private val GSON = GsonBuilder()
            .serializeNulls() //排除transient标记的字段
            .create()

    fun toJsonString(o: Any?): String {
        return GSON.toJson(o)
    }

    fun toJsonString(o: Any?, type: Type?): String {
        return GSON.toJson(o, type)
    }

    fun <T> fromJson(json: String?, clazz: Class<T>?): T? {
        return try {
            GSON.fromJson(json, clazz)
        } catch (e: JsonSyntaxException) {
            null
        }
    }

    fun <T> fromJson(json: String?, type: Type?): T? {
        return try {
            GSON.fromJson(json, type)
        } catch (e: JsonSyntaxException) {
            null
        }
    }

    fun <T> fromJson(json: JsonElement?, clazz: Class<T>?): T? {
        return try {
            GSON.fromJson(json, clazz)
        } catch (e: JsonSyntaxException) {
            null
        }
    }

    fun <T> fromJson(json: JsonElement?, type: Type?): T? {
        return try {
            GSON.fromJson(json, type)
        } catch (e: JsonSyntaxException) {
            null
        }
    }

    fun toJsonTree(o: Any?): JsonElement {
        return GSON.toJsonTree(o)
    }

    fun toJsonTree(o: Any?, type: Type?): JsonElement {
        return GSON.toJsonTree(o, type)
    }

    /**
     * 从json 中获取到value
     *
     * @param json
     * @param key
     * @return
     */
    fun getValue(json: String?, key: String?): JsonElement {
        return JsonParser.parseString(json).asJsonObject[key]
    }
}